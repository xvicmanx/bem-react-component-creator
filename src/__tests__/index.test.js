import React from 'react';
import renderer from 'react-test-renderer';
import BemComponentsCreator from '../index.js';

it('creates successfully a BEM based Widget', () => {
    const PostBEM = BemComponentsCreator('post');
    const Post = PostBEM.block('div');
    const Picture = PostBEM.element('img', 'picture');
    const Title = PostBEM.element('h2', 'title', 'awesome', 'large');
    const Phrase = PostBEM.element('p', 'phrase');

    const tree = renderer
    .create(
        <Post>
            <Picture
                src="https://http://via.placeholder.com/50x50"
                alt="Post illustration"
            />
            <Title>Test title</Title>
            <Phrase>
                An awesome phrase
            </Phrase>
            <Phrase modifiers="emphasized,colorful">
                An awesome phrase
            </Phrase>
        </Post>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
});