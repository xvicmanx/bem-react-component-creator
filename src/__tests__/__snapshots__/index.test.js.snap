// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creates successfully a BEM based Widget 1`] = `
<div
  className="post"
>
  <img
    alt="Post illustration"
    className="post__picture"
    src="https://http://via.placeholder.com/50x50"
  />
  <h2
    className="post__title post__title--awesome post__title--large"
  >
    Test title
  </h2>
  <p
    className="post__phrase"
  >
    An awesome phrase
  </p>
  <p
    className="post__phrase post__phrase--emphasized post__phrase--colorful"
  >
    An awesome phrase
  </p>
</div>
`;
